feat(scan): add a critical level to the severity

ostridm opened the pull request:
Introduce the ability to set the severity threshold to the `Critical` level using the `threshold` method:

```typescript
scan.threshold(Severity.CRITICAL);
```

With this implementation, the scan will continue despite the presence of issues with a severity lower than `Critical`, and will only stop for issues of `Critical` severity.

closes #129

ostridm committed:
fix(test): adjust failed test on Windows

ostridm committed:
feat(issue): extend `Severity` enum with `Critical` level

closes #129

derevnjuk commented on packages/reporter/src/std/StdReporter.ts:
```diff
@@ -9,6 +9,7 @@ export class StdReporter implements Reporter {
     Severity,
     (...x: unknown[]) => string
   > = {
+    [Severity.CRITICAL]: chalk.redBright,
```
Please add a new test case for logging critical severity issues to stderr. For details please refer to https://github.com/NeuraLegion/sectester-js/blob/bc05852b5252e71352a11e3f4e684f95641ab0f1/packages/reporter/src/std/StdReporter.spec.ts#L63

derevnjuk commented on packages/scan/src/models/Severity.spec.ts:
```diff
@@ -38,7 +38,8 @@ describe('Severity', () => {
       { input: Severity.MEDIUM, expected: 2 },
-      { input: Severity.HIGH, expected: 3 }
+      { input: Severity.HIGH, expected: 3 },
+      { input: Severity.CRITICAL, expected: 4 }
```
You have to add test data for `severityComparator`as well to cover missing cases

derevnjuk commented on packages/reporter/src/std/StdReporter.ts:
```diff
@@ -19,6 +20,7 @@ export class StdReporter implements Reporter {
     (...x: unknown[]) => void
   > = {
     /* eslint-disable no-console */
+    [Severity.CRITICAL]: console.error,
```
Critical issues are not being reported due to the code at https://github.com/NeuraLegion/sectester-js/blob/bc05852b5252e71352a11e3f4e684f95641ab0f1/packages/reporter/src/std/StdReporter.ts#L34

ostridm committed:
feat(scan): add a critical level to the severity

adjust according to code review:
https://github.com/NeuraLegion/sectester-js/pull/130#discussion_r1100573440
https://github.com/NeuraLegion/sectester-js/pull/130#discussion_r1100565873
https://github.com/NeuraLegion/sectester-js/pull/130#discussion_r1100563708

closes #129
